cpp/oneapi/dal/backend/primitives/test/ndarray.cpp

```diff
@@ -489,6 +503,42 @@ TEST_M(ndarray_test, "can assign ndarray", "[ndarray]") {
     sycl::free(x_ptr, queue);
 }
 
+TEST_M(ndarray_test, "can be flattened with host usm", "[ndarray]") {
+    DECLARE_TEST_POLICY(policy);
+    auto& queue = policy.get_queue();
+    auto [x, event] = ndarray<float, 2>::ones(queue, { 7, 5 }, sycl::usm::alloc::host);
+    event.wait_and_throw();
+
+    auto raw_arr = x.flatten();
+
+    for (std::int64_t i = 0; i < x.get_count(); ++i) {
+        REQUIRE(raw_arr[i] == 1.0f);
+    }
+}
+
+TEST_M(ndarray_test, "can be flattened with device usm", "[ndarray]") {
```

Comment:
Why we have different code to test host USM and device USM? We can have single code to test all three types of USM memory